lib/pull.ts

```diff
@@ -301,6 +303,60 @@ async function downloadAndSave(
 
     const meta = options ? options.meta : {};
 
+    const rootRequest = {
```

Comment:
All of the code from here through line 354 was pre-existing, and was just moved out of a nested function declaration